components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -883,6 +907,18 @@ void RRTMGPRadiation::run_impl (const double dt) {
       // Get IR 10.5 micron band for COSP
       auto idx_105 = rrtmgp::get_wavelength_index_lw(10.5e-6);
 
+      // Compute cloud-top diagnostics following AeroCOM recommendation
+      auto T_mid_at_cldtop       = real1d("T_mid_at_cldtop", ncol);
```

Comment:
Since these are all 1d arrays, why not making all these views use the pointer from the kokkos views? Something like
```
real1d T_mid_at_cldtop = real1d("T_mid_at_cldtop,d_T_mid_at_cldtop.data(),ncol);
```
This should allow you to remove the deep copies like `d_T_mid_at_cldtop(icol) = T_mid_at_cldtop(i+1);` below. Same goes for the cldlow/med/hgh/tot arrays...

Reply:
Agreed.